src/components/Projects/Project.tsx

```diff
@@ -1,80 +1,76 @@
-import { useState, useEffect } from "react";
+import { useEffect } from "react";
 import { NavLink } from "react-router-dom";
+import { GetRepoData, GetUserDetails } from "../../DATA/get_user_details";
+import { useAppContext } from "../../utils/Context";
+import { ActionTypes } from "../../utils/Reducers";
 
 type ProjectProps = {
   username: string;
   repo_name: string;
   name: string;
   link: string;
 };
-type RepoData = {
-  description: string;
-  language: string;
-};
-type LanguageInfo = { [key: string] : number };
 
 export default function Project({
   username,
   repo_name,
   name,
   link,
 }: ProjectProps) {
-  const [data, setData] = useState<RepoData | null>(null);
-  const [languagesData, setLanguagesData] = useState<LanguageInfo | null>(null);
+  const { state, dispatch } = useAppContext();
+  const repoData = state.repoData;
+  const userDetails = state.userDetails;
 
   useEffect(() => {
-    function getRepo() {
-      fetch(
-        `https://api.github.com/repos/${username.trim()}/${repo_name.trim()}?client_id=ec1594e91cfa6b4281cb&client_secret=02388e8e126c1f3d96d7b2a59350a3620c08c137`
-      )
-        .then((res) => res.json())
-        .then((result: RepoData) => setData(result));
+    async function getRepoData() {
+      try {
+        const result = await GetRepoData(username, repo_name);
+        dispatch({ type: ActionTypes.SET_REPO_DATA, payload: result });
+      } catch (error) {
+        throw new Error("Error while fetching the repos data");
+      }
     }
-    if (data) return;
-    getRepo();
+    getRepoData();
+  }, []);
 
-    function getLanguages() {
-      fetch(
-        `https://api.github.com/repos/${username.trim()}/${repo_name.trim()}/languages?client_id=ec1594e91cfa6b4281cb&client_secret=02388e8e126c1f3d96d7b2a59350a3620c08c137`
-      ).then((res) => res.json()).then((result: LanguageInfo) => {
-        setLanguagesData(result);
-      });
+  function fetchUserDetails() {
+    async function getUserDetails() {
+      try {
+        const result = await GetUserDetails(username);
+        dispatch({ type: ActionTypes.SET_DETAILS, payload: result });
+      } catch {
+        throw new Error("Failed to load user details");
+      }
     }
```

Comment:
no need to use this function here we can use it in user_info.tsx 

here we also need to use conditional fetching 
if user is present in the userDetails then no need to fetch him again else fetch him